front/src/router/index.ts

```diff
@@ -374,16 +376,29 @@ export function checkPathAvailable (needAuthentication: boolean, needGuest: bool
   return true
 }
 
+/**
+ * Check if the route accessible by roles
+ */
+export function checkRoles (userRole: USER_PROFILE | undefined, requiredRoles: USER_PROFILE[]) {
+  console.log('userRole', userRole)
+  return userRole && requiredRoles.includes(userRole)
+}
+
 router.beforeEach(function (to, from, next) {
   // To handle children routes (to get meta from parents), Vuejs recommend to use to.matched
   // @see: https://github.com/vuejs/vue-router/issues/704
   const needAuthentication = to.matched.some(m => m.meta.needAuthentication)
   appState.hasBurgerMenu = to.matched.some(m => m.meta.hasBurgerMenu)
   const needGuest = to.matched.some(m => m.meta.needGuest)
   const isAuthenticated = authState.data.isAuthenticated
+  console.log(to.matched)
```

Comment:
Here too ! You're "everywhere" !

![](https://media.giphy.com/media/kRXnZwKrPTwVq/giphy.gif)